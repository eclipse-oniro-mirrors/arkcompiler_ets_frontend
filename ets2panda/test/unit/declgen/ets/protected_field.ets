/*
 * Copyright (c) 2025 Huawei Device Co., Ltd.
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

export class C {
  public x: int = 7;
  protected s: string = "";
  private p?: Object;

  public foo(): void {};
  protected bar(x?: int) { return x!.toString(); };
  private baz(s: string) { console.println(s); };
}
