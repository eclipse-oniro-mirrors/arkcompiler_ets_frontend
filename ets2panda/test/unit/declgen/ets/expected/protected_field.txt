'use static'

export declare class C {
  public x: int;
  protected s: string;

  foo(): void;

  protected bar(x?: int): String;

  constructor();

}

